### ISSUE TITLE
CNI config 

### ISSUE THREAD

**Issue by mikebrow** [containerd/cri#979]
We should add the CNI configuration details to the runtime.StatusResponse when verbose is enabled. 

Will help us .. and customers debug issues like this one:
https://github.com/containerd/containerd/issues/2804

**Comment by mikebrow** [comment]
on second thought let's do this on inspectp / podsandboxstatus, since each pod may have it's own network configuration.

**Comment by Random-Liu** [comment]
> We should add the CNI configuration details to the runtime.StatusResponse when verbose is enabled.

This is also useful. Let's also do it. At least, we should list which exact CNI config file containerd is using.

**Comment by mikebrow** [comment]
i like that plan will push a commit if you don't beat me to it :)

**Comment by Random-Liu** [comment]
@mikebrow Then do it. :)

**Comment by abhi** [comment]
well go-cni config function will return the config at that moment. There is no guarantee that the same config was used for previously created  (and running) pods :)

**Comment by mikebrow** [comment]
yeah was going to expose the one used to attach.. with a new api from the go-cni api before we garbage collect it .. then stow it away .. sigh

**Comment by mikebrow** [comment]
Yeah .. what random said was already there cloning go-cni.. :)

**Comment by mikebrow** [comment]
@Random-Liu interesting so you think paths to configs and conflist.. hmm

**Comment by mikebrow** [comment]
Here's the new crictl inspectp output from a containerd/cri commit that I'll push when https://github.com/containerd/go-cni/pull/33 gets merged:

```
mike@mike-VirtualBox:~/go/src/crictl-test$ sudo crictl inspectp 46c
{
  "status": {
    "id": "46c14cc26e5f8c90de8b21d64b4f26bbe342f3d4ac0319cd2771dd6ff127abea",
    "metadata": {
      "attempt": 1,
      "name": "nginx-sandbox",
      "namespace": "default",
      "uid": "hdishd83djaidwnduwk28bcsb"
    },
    "state": "SANDBOX_READY",
    "createdAt": "2018-11-30T12:40:24.492124152-06:00",
    "network": {
      "ip": "10.88.10.229"
    },
    "linux": {
      "namespaces": {
        "options": {
          "ipc": "POD",
          "network": "POD",
          "pid": "POD"
        }
      }
    },
    "labels": {},
    "annotations": {}
  },
  "info": {
    "pid": 22045,
    "processStatus": "running",
    "netNamespaceClosed": false,
    "image": "gcr.io/google_containers/pause:3.1",
    "snapshotKey": "46c14cc26e5f8c90de8b21d64b4f26bbe342f3d4ac0319cd2771dd6ff127abea",
    "snapshotter": "overlayfs",
    "runtimeHandler": "",
    "runtimeType": "io.containerd.runtime.v1.linux",
    "runtimeOptions": {},
    "config": {
      "metadata": {
        "name": "nginx-sandbox",
        "uid": "hdishd83djaidwnduwk28bcsb",
        "namespace": "default",
        "attempt": 1
      },
      "linux": {}
    },
    "runtimeSpec": {
      "ociVersion": "1.0.1-dev",
      "process": {
        "user": {
          "uid": 0,
          "gid": 0
        },
        "args": [
          "/pause"
        ],
...
      "annotations": {
        "io.kubernetes.cri.container-type": "sandbox",
        "io.kubernetes.cri.sandbox-id": "46c14cc26e5f8c90de8b21d64b4f26bbe342f3d4ac0319cd2771dd6ff127abea"
... 
      },
    },
    "cniResult": {
      "Interfaces": {
        "cni0": {
          "IPConfigs": null,
          "Mac": "22:3f:a1:a1:ad:a2",
          "Sandbox": ""
        },
        "eth0": {
          "IPConfigs": [
            {
              "IP": "10.88.10.229",
              "Gateway": "10.88.0.1"
            }
          ],
          "Mac": "",
          "Sandbox": "/var/run/netns/cni-59afc846-f783-4d0c-73e5-a9266c2d77d7"
        },
        "vethfdefb653": {
          "IPConfigs": null,
          "Mac": "9a:ec:08:fc:4d:33",
          "Sandbox": ""
        }
      },
      "DNS": [
        {},
        {}
      ],
      "Routes": [
        {
          "dst": "0.0.0.0/0"
        }
      ]
    },
    "cniConfig": {
      "PluginDirs": [
        "/opt/cni/bin"
      ],
      "PluginConfDir": "/etc/cni/net.d",
      "Prefix": "eth",
      "Networks": [
        {
          "Config": {
            "Name": "cni-loopback",
            "CNIVersion": "0.3.1",
            "Plugins": [
              {
                "Network": {
                  "type": "loopback",
                  "ipam": {},
                  "dns": {}
                },
                "Source": "{\"type\":\"loopback\"}"
              }
            ],
            "Source": "{\n\"cniVersion\": \"0.3.1\",\n\"name\": \"cni-loopback\",\n\"plugins\": [{\n  \"type\": \"loopback\"\n}]\n}"
          },
          "IFName": "lo"
        },
        {
          "Config": {
            "Name": "containerd-net",
            "CNIVersion": "0.3.1",
            "Plugins": [
              {
                "Network": {
                  "type": "bridge",
                  "ipam": {
                    "type": "host-local"
                  },
                  "dns": {}
                },
                "Source": "{\"bridge\":\"cni0\",\"ipMasq\":true,\"ipam\":{\"routes\":[{\"dst\":\"0.0.0.0/0\"}],\"subnet\":\"10.88.0.0/16\",\"type\":\"host-local\"},\"isGateway\":true,\"promiscMode\":true,\"type\":\"bridge\"}"
              },
              {
                "Network": {
                  "type": "portmap",
                  "capabilities": {
                    "portMappings": true
                  },
                  "ipam": {},
                  "dns": {}
                },
                "Source": "{\"capabilities\":{\"portMappings\":true},\"type\":\"portmap\"}"
              }
            ],
            "Source": "{\n  \"cniVersion\": \"0.3.1\",\n  \"name\": \"containerd-net\",\n  \"plugins\": [\n    {\n      \"type\": \"bridge\",\n      \"bridge\": \"cni0\",\n      \"isGateway\": true,\n      \"ipMasq\": true,\n      \"promiscMode\": true,\n      \"ipam\": {\n        \"type\": \"host-local\",\n        \"subnet\": \"10.88.0.0/16\",\n        \"routes\": [\n          { \"dst\": \"0.0.0.0/0\" }\n        ]\n      }\n    },\n    {\n      \"type\": \"portmap\",\n      \"capabilities\": {\"portMappings\": true}\n    }\n  ]\n}\n"
          },
          "IFName": "eth0"
        }
      ]
    }
  }
}
```

**Comment by mikebrow** [comment]
Beyond this additional detail I'd like to capture the cni log during add so we can log more detail on failure.

**Comment by Random-Liu** [comment]
This is addressed in #1158
